apps/web-staking/src/app/components/navbar/NavbarComponent.tsx

```diff
@@ -29,11 +30,19 @@ export default function NavbarComponent() {
       : "";
 	}
 
+	const handleToggle = () => {
+		const currentTime = new Date().getTime();
+		if (currentTime - lastToggle > 500) {
+			setIsMenuOpen(!isMenuOpen);
+			setLastToggle(currentTime);
+		}
+	}
```

Comment:
Why do we need to handle the 500 ms for closing, this seems strange to have in here ?

Reply:
Without it, when the user clicks the hamburger button, the menu opens and then close button is instantly triggered from the same click.

I tried using `e.stopPropagation()` but still encountered the same behavior.

Reply:
As discussed this is Ok for now, please add the comment and crteate a chore in the icebox to clean this up.

Reply:
Comment added to code, chore created here => https://www.pivotaltracker.com/story/show/188527767

Thank you!